### DISCUSSION TITLE
MIDI over USB

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#16034] [STM32 / Pyboard]
Hi, 
I want to create a MIDI-over-USB device on a stm32 board. I see that the machine.USBDevice is implemented for RP2040 and SAMD which you can use with the [midi implementation available in the micropython-lib](https://github.com/micropython/micropython-lib/blob/master/micropython/usb/usb-device-midi/usb/device/midi.py).
Any guidelines how to port this to stm32?
Thanks

**Comment by jonnor** [comment]
Using the new USBDevice for MIDI would require USBDevice to be ported to STM32. It seems that has been discussed a bit at https://github.com/orgs/micropython/discussions/16060   - linking it here so that other travellers that search may find it.
